src/NHibernate/Loader/SelectMode.cs

```diff
@@ -0,0 +1,32 @@
+namespace NHibernate.Loader
+{
+	public enum SelectMode
+	{
+		/// <summary>
+		/// Default behavior (as if no select mode applied).
+		/// </summary>
+		Default,
+
+		/// <summary>
+		/// Fetch entity.
+		/// </summary>
+		Fetch,
+		
+		/// <summary>
+		/// Fetch lazy properties.
+		/// </summary>
+		FetchLazyProperties,
+
+		/// <summary>
+		/// Only Identifier columns are added to select statement.
+		/// Use it for fetching child objects for already loaded entities.
+		/// Note: Missing entities in session will be returned as null
+		/// </summary>
+		ChildFetch,
```

Comment:
> Note: Missing entities in session will be returned as null

What does that mean? That using this mode could result in persisted entities to be returned as null if they were not already loaded in the session? If yes, this is a very unusual behavior for NHibernate. It should yield a proxy instead. (Or throw an error, or trigger a load, whatever, but guarantee no inconsistent data is returned.)

Reply:
> You should be ready for null if you are doing right join for instance.

Yes, but that does not look to me as a right join case here.

There are not any cases in NHibernate where a query result correctness depends on having already loaded these entities in the session. Yielding `null` in the query results instead of an entity (or proxy of an entity) because this entity was not already loaded in the session is not acceptable in my opinion.

Reply:
It appears to me that yielding `null` will moreover wreck child collection loading, defeating the purpose of the feature. See `Loader.ReadCollectionElement`, which is called on the row array. It may end up calling `EntityType.ResolveIdentifier` which has an early exit bypassing instantiating the element if the owner is null.